--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -395,6 +395,15 @@
 #  define fsync _commit
 #endif  /* ! __WATCOMC__ || __QNX__ */
 
+#ifdef __APPLE__
+#  if TARGET_OS_IOS
+#    undef HAVE_SYSTEM
+#    undef HAVE_SENDFILE
+#  else
+#    define HAVE_SYSTEM 1
+#  endif
+#endif
+
 /*[clinic input]
 # one of the few times we lie about this name!
 module os
@@ -12497,10 +12506,15 @@
         Py_BEGIN_ALLOW_THREADS
 #ifdef HAVE_MKFIFOAT
         if (dir_fd != DEFAULT_DIR_FD) {
+// BLENDER: disable also at compile time for compatibility when linking with older Xcode.
+// https://github.com/python/cpython/issues/97897
+#ifndef __APPLE__
             if (HAVE_MKFIFOAT_RUNTIME) {
                 result = mkfifoat(dir_fd, path->narrow, mode);
 
-            } else {
+            } else
+#endif
+            {
                 mkfifoat_unavailable = 1;
                 result = 0;
             }
@@ -12566,10 +12580,15 @@
         Py_BEGIN_ALLOW_THREADS
 #ifdef HAVE_MKNODAT
         if (dir_fd != DEFAULT_DIR_FD) {
+// BLENDER: disable also at compile time for compatibility when linking with older Xcode.
+// https://github.com/python/cpython/issues/97897
+#ifndef __APPLE__
             if (HAVE_MKNODAT_RUNTIME) {
                 result = mknodat(dir_fd, path->narrow, mode, device);
 
-            } else {
+            } else
+#endif
+            {
                 mknodat_unavailable = 1;
                 result = 0;
             }
--- a/Modules/Setup
+++ b/Modules/Setup
@@ -130,77 +130,84 @@
 
 # Modules that should always be present (POSIX and Windows):
 
-#_asyncio _asynciomodule.c
-#_bisect _bisectmodule.c
-#_contextvars _contextvarsmodule.c
-#_csv _csv.c
-#_datetime _datetimemodule.c
-#_decimal _decimal/_decimal.c
-#_heapq _heapqmodule.c
+*static*
+_asyncio _asynciomodule.c
+_bisect _bisectmodule.c
+_contextvars _contextvarsmodule.c
+_csv _csv.c
+_datetime _datetimemodule.c
+_decimal _decimal/_decimal.c
+_heapq _heapqmodule.c
 #_interpchannels _interpchannelsmodule.c
 #_interpqueues _interpqueuesmodule.c
 #_interpreters _interpretersmodule.c
-#_json _json.c
-#_lsprof _lsprof.c rotatingtree.c
-#_multiprocessing -I$(srcdir)/Modules/_multiprocessing _multiprocessing/multiprocessing.c _multiprocessing/semaphore.c
-#_opcode _opcode.c
-#_pickle _pickle.c
-#_queue _queuemodule.c
-#_random _randommodule.c
-#_socket socketmodule.c
-#_statistics _statisticsmodule.c
-#_struct _struct.c
-#_typing _typingmodule.c
-#_zoneinfo _zoneinfo.c
-#array arraymodule.c
-#binascii binascii.c
-#cmath cmathmodule.c
-#math mathmodule.c
-#mmap mmapmodule.c
-#select selectmodule.c
+_json _json.c
+_lsprof _lsprof.c rotatingtree.c
+_multiprocessing -I$(srcdir)/Modules/_multiprocessing _multiprocessing/multiprocessing.c _multiprocessing/semaphore.c
+_opcode _opcode.c
+_pickle _pickle.c
+_queue _queuemodule.c
+_random _randommodule.c
+_socket socketmodule.c
+_statistics _statisticsmodule.c
+_struct _struct.c
+_typing _typingmodule.c
+_zoneinfo _zoneinfo.c
+array arraymodule.c
+binascii binascii.c
+cmath cmathmodule.c
+math mathmodule.c
+mmap mmapmodule.c
+select selectmodule.c
 #_sysconfig _sysconfig.c
 
 # XML
-#_elementtree _elementtree.c
-#pyexpat pyexpat.c
+*static*
+_elementtree _elementtree.c
+pyexpat pyexpat.c
 
 # hashing builtins
-#_blake2 _blake2/blake2module.c _blake2/blake2b_impl.c _blake2/blake2s_impl.c
-#_md5 md5module.c -I$(srcdir)/Modules/_hacl/include _hacl/Hacl_Hash_MD5.c -D_BSD_SOURCE -D_DEFAULT_SOURCE
-#_sha1 sha1module.c -I$(srcdir)/Modules/_hacl/include _hacl/Hacl_Hash_SHA1.c -D_BSD_SOURCE -D_DEFAULT_SOURCE
-#_sha2 sha2module.c -I$(srcdir)/Modules/_hacl/include Modules/_hacl/libHacl_Hash_SHA2.a
-#_sha3 sha3module.c -I$(srcdir)/Modules/_hacl/include _hacl/Hacl_Hash_SHA3.c -D_BSD_SOURCE -D_DEFAULT_SOURCE
+*static*
+_blake2 _blake2/blake2module.c _blake2/blake2b_impl.c _blake2/blake2s_impl.c
+_md5 md5module.c -I$(srcdir)/Modules/_hacl/include _hacl/Hacl_Hash_MD5.c -D_BSD_SOURCE -D_DEFAULT_SOURCE
+_sha1 sha1module.c -I$(srcdir)/Modules/_hacl/include _hacl/Hacl_Hash_SHA1.c -D_BSD_SOURCE -D_DEFAULT_SOURCE
+_sha2 sha2module.c -I$(srcdir)/Modules/_hacl/include Modules/_hacl/libHacl_Hash_SHA2.a
+_sha3 sha3module.c -I$(srcdir)/Modules/_hacl/include _hacl/Hacl_Hash_SHA3.c -D_BSD_SOURCE -D_DEFAULT_SOURCE
 
 # text encodings and unicode
-#_codecs_cn cjkcodecs/_codecs_cn.c
-#_codecs_hk cjkcodecs/_codecs_hk.c
-#_codecs_iso2022 cjkcodecs/_codecs_iso2022.c
-#_codecs_jp cjkcodecs/_codecs_jp.c
-#_codecs_kr cjkcodecs/_codecs_kr.c
-#_codecs_tw cjkcodecs/_codecs_tw.c
-#_multibytecodec cjkcodecs/multibytecodec.c
-#unicodedata unicodedata.c
+*static*
+_codecs_cn cjkcodecs/_codecs_cn.c
+_codecs_hk cjkcodecs/_codecs_hk.c
+_codecs_iso2022 cjkcodecs/_codecs_iso2022.c
+_codecs_jp cjkcodecs/_codecs_jp.c
+_codecs_kr cjkcodecs/_codecs_kr.c
+_codecs_tw cjkcodecs/_codecs_tw.c
+_multibytecodec cjkcodecs/multibytecodec.c
+unicodedata unicodedata.c
 
 # Modules with some UNIX dependencies
 
-#_posixsubprocess _posixsubprocess.c
-#_posixshmem -I$(srcdir)/Modules/_multiprocessing _multiprocessing/posixshmem.c -lrt
-#fcntl fcntlmodule.c
-#grp grpmodule.c
-#resource resource.c
-#syslog syslogmodule.c
-#termios termios.c
+*static*
+_posixsubprocess _posixsubprocess.c
+_posixshmem _multiprocessing/posixshmem.c
+fcntl fcntlmodule.c
+grp grpmodule.c
+resource resource.c
+syslog syslogmodule.c
+termios termios.c
 
 # Modules that require external libraries.
 
-#_bz2 _bz2module.c -lbz2
-#_ctypes _ctypes/_ctypes.c _ctypes/callbacks.c _ctypes/callproc.c _ctypes/stgdict.c _ctypes/cfield.c -ldl -lffi -DHAVE_FFI_PREP_CIF_VAR -DHAVE_FFI_PREP_CLOSURE_LOC -DHAVE_FFI_CLOSURE_ALLOC
+*static*
+_bz2 _bz2module.c -lbz2
+_ctypes _ctypes/_ctypes.c _ctypes/callbacks.c _ctypes/callproc.c _ctypes/stgdict.c _ctypes/cfield.c -ldl -lffi -DHAVE_FFI_PREP_CIF_VAR -DHAVE_FFI_PREP_CLOSURE_LOC -DHAVE_FFI_CLOSURE_ALLOC
 # The _dbm module supports NDBM, GDBM with compat module, and Berkeley DB.
 #_dbm _dbmmodule.c -lgdbm_compat -DUSE_GDBM_COMPAT
 #_gdbm _gdbmmodule.c -lgdbm
-#_lzma _lzmamodule.c -llzma
-#_uuid _uuidmodule.c -luuid
-#zlib  zlibmodule.c -lz
+*static*
+_lzma _lzmamodule.c -llzma
+_uuid _uuidmodule.c #-luuid
+zlib  zlibmodule.c -lz
 
 # The readline module also supports libeditline (-leditline).
 # Some systems may require -ltermcap or -ltermlib.
@@ -216,6 +223,9 @@
 #     -l:libcrypto.a -Wl,--exclude-libs,libcrypto.a
 # _hashlib _hashopenssl.c $(OPENSSL_INCLUDES) $(OPENSSL_LDFLAGS) \
 #     -l:libcrypto.a -Wl,--exclude-libs,libcrypto.a
+*static*
+_ssl _ssl.c $(OPENSSL_INCLUDES) $(OPENSSL_LDFLAGS) $(OPENSSL_LIBS)
+_hashlib _hashopenssl.c $(OPENSSL_INCLUDES) $(OPENSSL_LDFLAGS) -lcrypto
 
 # The _tkinter module.
 #
@@ -263,13 +273,15 @@
 #_curses_panel -lpanel -lncurses _curses_panel.c
 
 # macOS specific module, needs SystemConfiguration and CoreFoundation framework
-# _scproxy _scproxy.c
+*static*
+_scproxy _scproxy.c
 
 # Examples
 
 #xx xxmodule.c
-#xxlimited xxlimited.c
-#xxlimited_35 xxlimited_35.c
+*static*
+xxlimited xxlimited.c
+xxlimited_35 xxlimited_35.c
 #xxsubtype xxsubtype.c
 
 # Testing
@@ -294,5 +306,6 @@
 #
 #*disabled*
 #
-# _sqlite3 _tkinter _curses pyexpat
-# _codecs_jp _codecs_kr _codecs_tw unicodedata
+*static*
+_sqlite3 _tkinter _curses pyexpat
+_codecs_jp _codecs_kr _codecs_tw unicodedata
--- a/Modules/_scproxy.c
+++ b/Modules/_scproxy.c
@@ -60,6 +60,9 @@
 static PyObject*
 get_proxy_settings(PyObject* Py_UNUSED(mod), PyObject *Py_UNUSED(ignored))
 {
+#if defined(__APPLE__)
+  return PyDict_New();
+#else
     CFDictionaryRef proxyDict = NULL;
     CFNumberRef aNum = NULL;
     CFArrayRef anArray = NULL;
@@ -128,6 +131,7 @@
     if (proxyDict)  CFRelease(proxyDict);
     Py_XDECREF(result);
     return NULL;
+#endif
 }
 
 static int
@@ -172,6 +176,9 @@
 static PyObject*
 get_proxies(PyObject* Py_UNUSED(mod), PyObject *Py_UNUSED(ignored))
 {
+#if defined(__APPLE__)
+  return PyDict_New();
+#else
     PyObject* result = NULL;
     int r;
     CFDictionaryRef proxyDict = NULL;
@@ -219,6 +226,7 @@
     if (proxyDict)  CFRelease(proxyDict);
     Py_XDECREF(result);
     return NULL;
+#endif
 }
 
 static PyMethodDef mod_methods[] = {
