diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 864f5d5..c8d5844 100755
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -28,12 +28,12 @@
 message(STATUS "cmake version ${CMAKE_VERSION}")
 if(POLICY CMP0025)
-    cmake_policy(SET CMP0025 OLD) # report Apple's Clang as just Clang
+    cmake_policy(SET CMP0025 NEW) # report Apple's Clang as just Clang
 endif()
 if(POLICY CMP0042)
     cmake_policy(SET CMP0042 NEW) # MACOSX_RPATH
 endif()
 if(POLICY CMP0054)
-    cmake_policy(SET CMP0054 OLD) # Only interpret if() arguments as variables or keywords when unquoted
+    cmake_policy(SET CMP0054 NEW) # Only interpret if() arguments as variables or keywords when unquoted
 endif()
 
 project (x265)
@@ -522,6 +522,10 @@ if(ARM64)
     list(APPEND ASM_FLAGS -DHIGH_BIT_DEPTH=0 -DBIT_DEPTH=8 -DX265_NS=${X265_NS})
   endif()
 endif(ARM64)
 
+if(APPLE)
+  list(APPEND ASM_FLAGS -isysroot ${CMAKE_OSX_SYSROOT} ${APPLE_OS_MINVERSION_CFLAG})
+endif()
+
 if(ENABLE_PIC)
     if(XCODE)
         append_property(GET XC BB_GENERATE_PIC "YES")
