diff --git a/source/blender/blentranslation/msgfmt/CMakeLists.txt b/source/blender/blentranslation/msgfmt/CMakeLists.txt
index 8300f1bfcf0..10c71340476 100644
--- a/source/blender/blentranslation/msgfmt/CMakeLists.txt
+++ b/source/blender/blentranslation/msgfmt/CMakeLists.txt
@@ -30,14 +30,16 @@ if(WIN32)
   list(APPEND LIB bf_intern_utfconv)
 endif()
 
-add_executable(msgfmt ${SRC})
-setup_platform_linker_flags(msgfmt)
-blender_target_include_dirs(msgfmt ${INC})
-target_link_libraries(msgfmt ${LIB})
-
-# If using an external manifest, copy it to the output directory after building.
-if(WITH_WINDOWS_EXTERNAL_MANIFEST)
-  add_custom_command(TARGET msgfmt POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_BINARY_DIR}/tests.exe.manifest  $<TARGET_FILE_DIR:msgfmt>/$<TARGET_FILE_NAME:msgfmt>.manifest
-  )
+if(NOT WITH_CROSSCOMPILED_TOOLS)
+  add_executable(msgfmt ${SRC})
+  setup_platform_linker_flags(msgfmt)
+  blender_target_include_dirs(msgfmt ${INC})
+  target_link_libraries(msgfmt ${LIB})
+
+  # If using an external manifest, copy it to the output directory after building.
+  if(WITH_WINDOWS_EXTERNAL_MANIFEST)
+    add_custom_command(TARGET msgfmt POST_BUILD
+      COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_BINARY_DIR}/tests.exe.manifest  $<TARGET_FILE_DIR:msgfmt>/$<TARGET_FILE_NAME:msgfmt >.manifest
+    )
+  endif()
 endif()
