diff --git a/source/blender/gpu/intern/gpu_framebuffer_private.hh b/source/blender/gpu/intern/gpu_framebuffer_private.hh
index 0711428a30a..7bc59531ed6 100644
--- a/source/blender/gpu/intern/gpu_framebuffer_private.hh
+++ b/source/blender/gpu/intern/gpu_framebuffer_private.hh
@@ -174,6 +174,19 @@ class FrameBuffer {
     if (!equals_v4v4_int(viewport_[0], viewport)) {
       copy_v4_v4_int(viewport_[0], viewport);
       dirty_state_ = true;
+#if (WITH_APPLE_CROSSPLATFORM)
+      /* IOS_FIXME - selecting 2D Full Canvas window causes a viewport with a -1 origin.
+       Workaround for now. */
+      if (viewport_[0][0] < 0 || viewport_[0][1] < 0) {
+        printf("Invalid viewport detected: %d,%d - %dx%d\n",
+               viewport_[0][0],
+               viewport_[0][1],
+               viewport_[0][2],
+               viewport_[0][3]);
+        viewport_[0][0] = max_ii(viewport_[0][0], 0);
+        viewport_[0][1] = max_ii(viewport_[0][1], 0);
+      }
+#endif
     }
     multi_viewport_ = false;
   }
