diff --git a/intern/ghost/GHOST_ISystem.hh b/intern/ghost/GHOST_ISystem.hh
index 810b4d0418e..1b4336f76c5 100644
--- a/intern/ghost/GHOST_ISystem.hh
+++ b/intern/ghost/GHOST_ISystem.hh
@@ -464,6 +464,18 @@ class GHOST_ISystem {
    */
   virtual void putClipboard(const char *buffer, bool selection) const = 0;
 
+#if (WITH_APPLE_CROSSPLATFORM)
+  virtual GHOST_TSuccess popupOnScreenKeyboard(
+      GHOST_IWindow *window, const GHOST_KeyboardProperties &keyboard_properties) = 0;
+
+  virtual GHOST_TSuccess hideOnScreenKeyboard(GHOST_IWindow *window) = 0;
+
+  virtual const char *getKeyboardInput(GHOST_IWindow *window) = 0;
+
+  virtual GHOST_TSuccess startSecurityScopedFileAccess(const char *filepath) = 0;
+  virtual GHOST_TSuccess stopSecurityScopedFileAccess(const char *filepath) = 0;
+#endif
+
   /**
    * Returns GHOST_kSuccess if the clipboard contains an image.
    */
