diff --git a/build_files/build_environment/cmake/fftw.cmake b/build_files/build_environment/cmake/fftw.cmake
index 4b58aa62215..9186714c6c9 100644
--- a/build_files/build_environment/cmake/fftw.cmake
+++ b/build_files/build_environment/cmake/fftw.cmake
@@ -27,6 +27,12 @@ macro(fftw_build FFTW_POSTFIX)
       INSTALL_DIR ${LIBDIR}/fftw3
     )
   else()
+    if(WITH_APPLE_CROSSPLATFORM)
+      # Building for non-local architecture.
+      set(CROSS_COMPILE_FLAGS "--host=arm")
+    else()
+      set(CROSS_COMPILE_FLAGS)
+    endif()
     set(FFTW_EXTRA_ARGS --enable-static --enable-threads)
     set(FFTW_INSTALL install)
     ExternalProject_Add(external_fftw3_${FFTW_POSTFIX}
@@ -38,7 +44,8 @@ macro(fftw_build FFTW_POSTFIX)
       CONFIGURE_COMMAND ${CONFIGURE_ENV} &&
         cd ${BUILD_DIR}/fftw3/src/external_fftw3_${FFTW_POSTFIX}/ &&
         ${CONFIGURE_COMMAND} ${FFTW_EXTRA_ARGS} ${ARGN} --prefix=${mingw_LIBDIR}/fftw3
-
+        ${CROSS_COMPILE_FLAGS}
+      
       BUILD_COMMAND ${CONFIGURE_ENV} &&
         cd ${BUILD_DIR}/fftw3/src/external_fftw3_${FFTW_POSTFIX}/ &&
         make -j${MAKE_THREADS}
