diff --git a/build_files/build_environment/cmake/lame.cmake b/build_files/build_environment/cmake/lame.cmake
index 7d30f40f365..68c574c2dc3 100644
--- a/build_files/build_environment/cmake/lame.cmake
+++ b/build_files/build_environment/cmake/lame.cmake
@@ -24,13 +24,20 @@ if(MSVC)
       ${LIBDIR}/lame/lib/mp3lame.lib
   )
 else()
-  set(LAME_CONFIGURE
-    ${CONFIGURE_ENV} &&
-    cd ${BUILD_DIR}/lame/src/external_lame/ &&
-    ${CONFIGURE_COMMAND}
-      --prefix=${LIBDIR}/lame
-      --disable-shared
-      --enable-static
+  if(WITH_APPLE_CROSSPLATFORM)
+    # Building for non-local architecture.
+    set(CROSS_COMPILE_FLAGS "--host=arm")
+  else()
+    set(CROSS_COMPILE_FLAGS)
+  endif()
+
+  set(LAME_CONFIGURE 
+    ${CONFIGURE_ENV} && 
+    cd ${BUILD_DIR}/lame/src/external_lame/ && 
+    ${CONFIGURE_COMMAND} 
+      --prefix=${LIBDIR}/lame 
+      --disable-shared 
+      --enable-static 
       ${LAME_EXTRA_ARGS}
       --enable-export=full
       --with-fileio=sndfile
@@ -39,7 +46,8 @@ else()
       --disable-mp3x
       --disable-mp3rtp
       --disable-gtktest
-      --disable-frontend)
+      --disable-frontend
+      ${CROSS_COMPILE_FLAGS})
   set(LAME_BUILD ${CONFIGURE_ENV} &&
     cd ${BUILD_DIR}/lame/src/external_lame/ &&
     make -j${MAKE_THREADS}
