diff --git a/build_files/build_environment/cmake/openvdb.cmake b/build_files/build_environment/cmake/openvdb.cmake
index 514121e7efe..1065f869d7d 100644
--- a/build_files/build_environment/cmake/openvdb.cmake
+++ b/build_files/build_environment/cmake/openvdb.cmake
@@ -47,6 +47,31 @@ set(OPENVDB_EXTRA_ARGS
   # -DLLVM_DIR=${LIBDIR}/llvm/lib/cmake/llvm
 )
 
+if(WITH_APPLE_CROSSPLATFORM) 
+  # IOS_FIXME: Think we can remove this now Boost has been removed
+  file(GLOB Boost_LIBRARIES
+    "${LIBDIR}/boost/lib/*.dylib"
+  )
+
+  set(OPENVDB_EXTRA_ARGS
+    ${OPENVDB_EXTRA_ARGS}
+    #-DCMAKE_POLICY_DEFAULT_CMP0144:STRING=NEW 
+    # TEMP: Disable python module as library discovery fails.
+    -DOPENVDB_BUILD_PYTHON_MODULE=OFF
+
+    # Boost
+    -DBoost_ROOT=${LIBDIR}/boost
+    #-DBoost_FOUND=YES
+    #-DBoost_VERSION=${BOOST_VERSION}
+    -DBoost_INCLUDE_DIR=${LIBDIR}/boost/include/
+    #-DBoost_LIBRARIES=${Boost_LIBRARIES}
+    #-DBoost_LIBRARY_DIRS=${LIBDIR}/boost/lib/
+    -DBoost_DEBUG=ON
+    #-DBoost_LIB_PREFIX=lib
+  )
+endif()
+
+
 set(OPENVDB_PATCH
   ${PATCH_CMD} -p 1 -d
     ${BUILD_DIR}/openvdb/src/openvdb <
