diff --git a/build_files/build_environment/cmake/opensubdiv.cmake b/build_files/build_environment/cmake/opensubdiv.cmake
index 16788cf8233..5160ade7272 100644
--- a/build_files/build_environment/cmake/opensubdiv.cmake
+++ b/build_files/build_environment/cmake/opensubdiv.cmake
@@ -2,7 +2,21 @@
 #
 # SPDX-License-Identifier: GPL-2.0-or-later
 
+if(WITH_APPLE_CROSSPLATFORM)
+  # Disable OpenGL and TBB for IOS
+  set(SUBDIV_PLATFORM_FLAGS
+    -DNO_OPENGL=ON
+    -DNO_TBB=ON
+  )
+else()
+  set(SUBDIV_PLATFORM_FLAGS
+    -DNO_OPENGL=OFF
+    -DNO_TBB=OFF
+  )
+endif()
+
 set(OPENSUBDIV_EXTRA_ARGS
+${SUBDIV_PLATFORM_FLAGS}
   -DNO_LIB=OFF
   -DNO_EXAMPLES=ON
   -DNO_TUTORIALS=ON
@@ -10,11 +24,9 @@ set(OPENSUBDIV_EXTRA_ARGS
   -DNO_PTEX=ON
   -DNO_DOC=ON
   -DNO_OMP=ON
-  -DNO_TBB=OFF
   -DNO_CUDA=ON
   -DNO_OPENCL=ON
   -DNO_CLEW=ON
-  -DNO_OPENGL=OFF
   -DNO_METAL=OFF
   -DNO_DX=ON
   -DNO_TESTS=ON
@@ -72,6 +84,7 @@ if(WIN32)
   endif()
 else()
   harvest(external_opensubdiv opensubdiv/include opensubdiv/include "*.h")
+  harvest(external_opensubdiv opensubdiv/lib opensubdiv/lib "*.a")
   harvest_rpath_lib(external_opensubdiv opensubdiv/lib opensubdiv/lib "*${SHAREDLIBEXT}*")
 endif()
 
