diff --git a/build_files/build_environment/cmake/opus.cmake b/build_files/build_environment/cmake/opus.cmake
index 16f8d159c2a..c1d85f16a3b 100644
--- a/build_files/build_environment/cmake/opus.cmake
+++ b/build_files/build_environment/cmake/opus.cmake
@@ -2,6 +2,13 @@
 #
 # SPDX-License-Identifier: GPL-2.0-or-later
 
+if(WITH_APPLE_CROSSPLATFORM)
+  # Building for non-local architecture.
+  set(CROSS_COMPILE_FLAGS "--host=arm")
+else()
+  set(CROSS_COMPILE_FLAGS)
+endif()
+
 if(MSVC)
   set(OPUS_CMAKE_ARGS
     -DPACKAGE_VERSION=${OPUS_VERSION}
@@ -25,6 +32,7 @@ if(NOT WIN32)
         --enable-static
         --with-pic
         --disable-maintainer-mode
+	${CROSS_COMPILE_FLAGS}
 
     BUILD_COMMAND ${CONFIGURE_ENV} &&
       cd ${BUILD_DIR}/opus/src/external_opus/ &&
