diff --git a/build_files/build_environment/cmake/ssl.cmake b/build_files/build_environment/cmake/ssl.cmake
index f2c3043682d..99c6cac2d3f 100644
--- a/build_files/build_environment/cmake/ssl.cmake
+++ b/build_files/build_environment/cmake/ssl.cmake
@@ -66,3 +66,8 @@ else()
     INSTALL_DIR ${LIBDIR}/ssl
   )
 endif()
+
+if(WITH_APPLE_CROSSPLATFORM)
+  # Required to provide libs for IOS_PYTHON_STATIC_LIBS
+  harvest_rpath_lib(external_ssl ssl/lib ssl/lib "*.a")
+endif()
