diff --git a/build_files/build_environment/cmake/xml2.cmake b/build_files/build_environment/cmake/xml2.cmake
index 14d94947585..1575144912f 100644
--- a/build_files/build_environment/cmake/xml2.cmake
+++ b/build_files/build_environment/cmake/xml2.cmake
@@ -11,11 +11,26 @@ set(XML2_EXTRA_ARGS
   -DLIBXML2_WITH_PROGRAMS=OFF
   -DBUILD_SHARED_LIBS=OFF
 )
+
+if(WITH_APPLE_CROSSPLATFORM)
+  # Patch to remove use of getEntropy()
+  set(APPLE_CROSSPLATFORM_PATCH_CMD 
+    ${PATCH_CMD} --verbose -p 1 -N -d 
+    ${BUILD_DIR}/xml2/src/external_xml2 < 
+    ${PATCH_DIR}/xml2_ios.diff
+  )
+else()
+  set(APPLE_CROSSPLATFORM_PATCH_CMD)
+endif()
+
 ExternalProject_Add(external_xml2
   URL file://${PACKAGE_DIR}/${XML2_FILE}
   DOWNLOAD_DIR ${DOWNLOAD_DIR}
   URL_HASH ${XML2_HASH_TYPE}=${XML2_HASH}
 
+  PATCH_COMMAND
+    ${APPLE_CROSSPLATFORM_PATCH_CMD}
+  
   CMAKE_ARGS
     -DCMAKE_INSTALL_PREFIX=${LIBDIR}/xml2
     ${DEFAULT_CMAKE_FLAGS}
