diff --git a/build_files/build_environment/cmake/zlib.cmake b/build_files/build_environment/cmake/zlib.cmake
index ba2d89d6a17..53e7f68588c 100644
--- a/build_files/build_environment/cmake/zlib.cmake
+++ b/build_files/build_environment/cmake/zlib.cmake
@@ -67,4 +67,8 @@ endif()
 if(UNIX AND NOT APPLE)
   harvest(external_zlib zlib/include zlib/include "*.h")
   harvest(external_zlib zlib/lib zlib/lib "*.a")
+elseif(WITH_APPLE_CROSSPLATFORM)
+  harvest(external_zlib zlib/include zlib/include "*.h")
+  harvest(external_zlib zlib/lib zlib/lib "*.a")
+  harvest(external_zlib zlib/lib zlib/lib "*.dylib")  
 endif()
