diff --git a/source/blender/makesrna/intern/CMakeLists.txt b/source/blender/makesrna/intern/CMakeLists.txt
index 70134166ece..83f88803975 100644
--- a/source/blender/makesrna/intern/CMakeLists.txt
+++ b/source/blender/makesrna/intern/CMakeLists.txt
@@ -413,23 +413,25 @@ if(WITH_GMP)
 endif()
 
 # Build `makesrna` executable.
-add_cc_flags_custom_test(makesrna)
-
-add_executable(makesrna ${SRC} ${SRC_RNA_INC} ${SRC_DNA_INC})
-setup_platform_linker_flags(makesrna)
-setup_platform_linker_libs(makesrna)
-blender_target_include_dirs(makesrna ${INC})
-blender_target_include_dirs_sys(makesrna ${INC_SYS})
-
-target_link_libraries(makesrna PRIVATE bf_dna)
-target_link_libraries(makesrna PRIVATE bf::intern::atomic)
-target_link_libraries(makesrna PRIVATE bf::intern::guardedalloc)
-target_link_libraries(makesrna PRIVATE bf_dna_blenlib)
-target_link_libraries(makesrna PRIVATE bf::dependencies::optional::tbb)
-
-if(WIN32 AND NOT UNIX)
-  if(DEFINED PTHREADS_LIBRARIES)
-    target_link_libraries(makesrna PRIVATE ${PTHREADS_LIBRARIES})
+if(NOT WITH_CROSSCOMPILED_TOOLS)
+  add_cc_flags_custom_test(makesrna)
+
+  add_executable(makesrna ${SRC} ${SRC_RNA_INC} ${SRC_DNA_INC})
+  setup_platform_linker_flags(makesrna)
+  setup_platform_linker_libs(makesrna)
+  blender_target_include_dirs(makesrna ${INC})
+  blender_target_include_dirs_sys(makesrna ${INC_SYS})
+
+  target_link_libraries(makesrna PRIVATE bf_dna)
+  target_link_libraries(makesrna PRIVATE bf::intern::atomic)
+  target_link_libraries(makesrna PRIVATE bf::intern::guardedalloc)
+  target_link_libraries(makesrna PRIVATE bf_dna_blenlib)
+  target_link_libraries(makesrna PRIVATE bf::dependencies::optional::tbb)
+
+  if(WIN32 AND NOT UNIX)
+    if(DEFINED PTHREADS_LIBRARIES)
+      target_link_libraries(makesrna PRIVATE ${PTHREADS_LIBRARIES})
+    endif()
   endif()
 endif()
 
