diff --git a/source/blender/makesrna/intern/rna_wm.cc b/source/blender/makesrna/intern/rna_wm.cc
index f6811f30484..93050ee5076 100644
--- a/source/blender/makesrna/intern/rna_wm.cc
+++ b/source/blender/makesrna/intern/rna_wm.cc
@@ -192,6 +192,15 @@ static const EnumPropertyItem event_ndof_type_items[] = {
 #  endif
     {0, nullptr, 0, nullptr, nullptr},
 };
+
+static const EnumPropertyItem event_touch_type_items[] = {
+    {TOUCH_EDGE_SWIPE_IN_LEFT, "TOUCH_EDGE_SWIPE_IN_LEFT", 0, "Edge Swap In Left"},
+    {TOUCH_EDGE_SWIPE_IN_RIGHT, "TOUCH_EDGE_SWIPE_IN_RIGHT", 0, "Edge Swap In Right"},
+    {TOUCH_TWO_FINGER_TAP, "TOUCH_TWO_FINGER_TAP", 0, "2 Fingers Tap", ""},
+    {TOUCH_THREE_FINGER_TAP, "TOUCH_THREE_FINGER_TAP", 0, "3 Fingers Tap", ""},
+    {TOUCH_FOUR_FINGER_TAP, "TOUCH_FOUR_FINGER_TAP", 0, "4 Fingers Tap", ""},
+    {0, nullptr, 0, nullptr, nullptr},
+};
 #endif /* RNA_RUNTIME */
 
 /**
@@ -452,6 +461,12 @@ const EnumPropertyItem rna_enum_event_type_items[] = {
     {NDOF_BUTTON_10, "NDOF_BUTTON_10", 0, "NDOF Button 10", "NdofB10"},
     {NDOF_BUTTON_11, "NDOF_BUTTON_11", 0, "NDOF Button 11", "NdofB11"},
     {NDOF_BUTTON_12, "NDOF_BUTTON_12", 0, "NDOF Button 12", "NdofB12"},
+    /* Touch events. */
+    {TOUCH_EDGE_SWIPE_IN_LEFT, "TOUCH_EDGE_SWIPE_IN_LEFT", 0, "Swap |→   |"},
+    {TOUCH_EDGE_SWIPE_IN_RIGHT, "TOUCH_EDGE_SWIPE_IN_RIGHT", 0, "Swap |   ←|"},
+    {TOUCH_TWO_FINGER_TAP, "TOUCH_TWO_FINGER_TAP", 0, "2 Fingers Tap"},
+    {TOUCH_THREE_FINGER_TAP, "TOUCH_THREE_FINGER_TAP", 0, "3 Fingers Tap"},
+    {TOUCH_FOUR_FINGER_TAP, "TOUCH_FOUR_FINGER_TAP", 0, "4 Fingers Tap"},
 
     /* Action Zones. */
     {EVT_ACTIONZONE_AREA, "ACTIONZONE_AREA", 0, "ActionZone Area", "AZone Area"},
@@ -1137,6 +1152,10 @@ static void rna_wmKeyMapItem_map_type_set(PointerRNA *ptr, int value)
         kmi->type = NDOF_MOTION;
         kmi->val = KM_NOTHING;
         break;
+      case KMI_TYPE_TOUCH:
+        kmi->type = TOUCH_TWO_FINGER_TAP;
+        kmi->val = KM_NOTHING;
+        break;
     }
   }
 }
@@ -1179,6 +1198,9 @@ static const EnumPropertyItem *rna_KeyMapItem_type_itemf(bContext * /*C*/,
   if (map_type == KMI_TYPE_NDOF) {
     return event_ndof_type_items;
   }
+  if (map_type == KMI_TYPE_TOUCH) {
+    return event_touch_type_items;
+  }
   if (map_type == KMI_TYPE_TEXTINPUT) {
     return event_textinput_type_items;
   }
@@ -2989,6 +3011,7 @@ static void rna_def_keyconfig(BlenderRNA *brna)
   static const EnumPropertyItem map_type_items[] = {
       {KMI_TYPE_KEYBOARD, "KEYBOARD", 0, "Keyboard", ""},
       {KMI_TYPE_MOUSE, "MOUSE", 0, "Mouse", ""},
+      {KMI_TYPE_TOUCH, "TOUCH", 0, "Touch", ""},
       {KMI_TYPE_NDOF, "NDOF", 0, "NDOF", ""},
       {KMI_TYPE_TEXTINPUT, "TEXTINPUT", 0, "Text Input", ""},
       {KMI_TYPE_TIMER, "TIMER", 0, "Timer", ""},
