diff --git a/intern/ghost/intern/GHOST_SystemPathsCocoa.mm b/intern/ghost/intern/GHOST_SystemPathsCocoa.mm
index 762e0726b79..db8ec09ad84 100644
--- a/intern/ghost/intern/GHOST_SystemPathsCocoa.mm
+++ b/intern/ghost/intern/GHOST_SystemPathsCocoa.mm
@@ -2,7 +2,9 @@
  *
  * SPDX-License-Identifier: GPL-2.0-or-later */
 
-#import <AppKit/NSDocumentController.h>
+#ifndef WITH_APPLE_CROSSPLATFORM
+#  import <AppKit/NSDocumentController.h>
+#endif
 #import <Foundation/Foundation.h>
 
 #include <optional>
@@ -127,8 +129,13 @@ const char *GHOST_SystemPathsCocoa::getBinaryDir() const
 
 void GHOST_SystemPathsCocoa::addToSystemRecentFiles(const char *filepath) const
 {
+#ifndef WITH_APPLE_CROSSPLATFORM
   @autoreleasepool {
-    NSURL *file_url = [NSURL fileURLWithPath:[NSString stringWithUTF8String:filepath]];
+    NSURL *const file_url = [NSURL fileURLWithPath:[NSString stringWithUTF8String:filepath]];
     [[NSDocumentController sharedDocumentController] noteNewRecentDocumentURL:file_url];
   }
+#else
+  // Remove unused param warning
+  (void)(filepath);
+#endif
 }
