diff --git a/intern/ghost/intern/GHOST_C-api.cc b/intern/ghost/intern/GHOST_C-api.cc
index c10e0b20f24..efdb420f44a 100644
--- a/intern/ghost/intern/GHOST_C-api.cc
+++ b/intern/ghost/intern/GHOST_C-api.cc
@@ -70,6 +70,46 @@ void GHOST_ShowMessageBox(GHOST_SystemHandle systemhandle,
   system->showMessageBox(title, message, help_label, continue_label, link, dialog_options);
 }
 
+#if (WITH_APPLE_CROSSPLATFORM)
+void GHOST_popupOnScreenKeyboard(GHOST_WindowHandle windowhandle,
+                                 const GHOST_KeyboardProperties &keyboard_properties)
+{
+  GHOST_ISystem *system = GHOST_ISystem::getSystem();
+  GHOST_IWindow *window = (GHOST_IWindow *)windowhandle;
+
+  system->popupOnScreenKeyboard(window, keyboard_properties);
+}
+
+void GHOST_hideOnScreenKeyboard(GHOST_WindowHandle windowhandle)
+{
+  GHOST_ISystem *system = GHOST_ISystem::getSystem();
+  GHOST_IWindow *window = (GHOST_IWindow *)windowhandle;
+
+  system->hideOnScreenKeyboard(window);
+}
+
+const char *GHOST_getKeyboardInput(GHOST_WindowHandle windowhandle)
+{
+  GHOST_ISystem *system = GHOST_ISystem::getSystem();
+  GHOST_IWindow *window = (GHOST_IWindow *)windowhandle;
+
+  return system->getKeyboardInput(window);
+}
+
+extern GHOST_TSuccess GHOST_startSecurityScopedFileAccess(const char *filepath)
+{
+  GHOST_ISystem *system = GHOST_ISystem::getSystem();
+  return system->startSecurityScopedFileAccess(filepath);
+}
+
+extern GHOST_TSuccess GHOST_stopSecurityScopedFileAccess(const char *filepath)
+{
+  GHOST_ISystem *system = GHOST_ISystem::getSystem();
+  return system->stopSecurityScopedFileAccess(filepath);
+}
+
+#endif
+
 GHOST_EventConsumerHandle GHOST_CreateEventConsumer(GHOST_EventCallbackProcPtr eventCallback,
                                                     GHOST_TUserDataPtr user_data)
 {
