diff --git a/build_files/build_environment/cmake/alembic.cmake b/build_files/build_environment/cmake/alembic.cmake
index 2a6cb7630f6..09e4c0e0c81 100644
--- a/build_files/build_environment/cmake/alembic.cmake
+++ b/build_files/build_environment/cmake/alembic.cmake
@@ -15,6 +15,14 @@ set(ALEMBIC_EXTRA_ARGS
   -DALEMBIC_SHARED_LIBS=OFF
 )
 
+# Could not find Imath libs on its own so specify path
+if(WITH_APPLE_CROSSPLATFORM)
+  set(ALEMBIC_EXTRA_ARGS
+    ${ALEMBIC_EXTRA_ARGS}
+    -DImath_DIR=${LIBDIR}/imath/lib/cmake/Imath
+  )
+endif()
+
 ExternalProject_Add(external_alembic
   URL file://${PACKAGE_DIR}/${ALEMBIC_FILE}
   DOWNLOAD_DIR ${DOWNLOAD_DIR}
