diff --git a/build_files/build_environment/cmake/opencolorio.cmake b/build_files/build_environment/cmake/opencolorio.cmake
index 06314bce301..e7e488278b8 100644
--- a/build_files/build_environment/cmake/opencolorio.cmake
+++ b/build_files/build_environment/cmake/opencolorio.cmake
@@ -31,6 +31,19 @@ set(OPENCOLORIO_EXTRA_ARGS
   -Dpybind11_ROOT=${LIBDIR}/pybind11
 )
 
+if(WITH_APPLE_CROSSPLATFORM)
+  # Use iOS utility to set some env vars to help us build for iOS
+  include(cmake/ios_defines.cmake)
+  ios_get_dependency_env_vars(IMATH)
+  
+  set(OPENCOLORIO_EXTRA_ARGS
+    ${OPENCOLORIO_EXTRA_ARGS}
+    -DOCIO_BUILD_PYTHON=OFF
+    -DOCIO_USE_HEADLESS=ON
+    ${IOSDEP_DEFINES}
+  )
+endif()
+
 if(APPLE)
   set(OPENCOLORIO_EXTRA_ARGS
     ${OPENCOLORIO_EXTRA_ARGS}
@@ -74,6 +87,12 @@ else()
   )
 endif()
 
+if(WITH_APPLE_CROSSPLATFORM)
+  set(OCIO_PATCH PATCH_COMMAND ${PATCH_CMD} -p 1 -N -b -d ${BUILD_DIR}/opencolorio/src/external_opencolorio < ${PATCH_DIR}/opencolorio_ios.diff)
+else()
+  set(OCIO_PATCH)
+endif()
+
 ExternalProject_Add(external_opencolorio
   URL file://${PACKAGE_DIR}/${OPENCOLORIO_FILE}
   DOWNLOAD_DIR ${DOWNLOAD_DIR}
@@ -150,10 +169,13 @@ else()
 
   harvest(external_opencolorio opencolorio/include opencolorio/include "*.h")
   harvest_rpath_lib(external_opencolorio opencolorio/lib opencolorio/lib "*${SHAREDLIBEXT}*")
-  harvest_rpath_python(
-    external_opencolorio
-    opencolorio/lib/python${PYTHON_SHORT_VERSION}
-    python/lib/python${PYTHON_SHORT_VERSION}
-    "*"
-  )
+  # OCIO_BUILD_PYTHON currently set to OFF for iOS
+  if(NOT WITH_APPLE_CROSSPLATFORM)
+    harvest_rpath_python(
+      external_opencolorio
+      opencolorio/lib/python${PYTHON_SHORT_VERSION}
+      python/lib/python${PYTHON_SHORT_VERSION}
+      "*"
+    )
+  endif()
 endif()
