diff --git a/build_files/build_environment/cmake/sqlite.cmake b/build_files/build_environment/cmake/sqlite.cmake
index 57c3f0fb97f..fef3355fc19 100644
--- a/build_files/build_environment/cmake/sqlite.cmake
+++ b/build_files/build_environment/cmake/sqlite.cmake
@@ -5,10 +5,23 @@
 set(SQLITE_CONFIGURE_ENV echo .)
 set(SQLITE_CONFIGURATION_ARGS)
 
+if(WITH_APPLE_CROSSPLATFORM)
+  # Building for non-local architecture.
+  set(CROSS_COMPILE_FLAGS "--host=arm")
+else()
+  set(CROSS_COMPILE_FLAGS)
+endif()
+
 if(UNIX)
   if(NOT APPLE)
     set(SQLITE_LDFLAGS -Wl,--as-needed)
   endif()
+
+if(WITH_APPLE_CROSSPLATFORM)
+  # Flags from configure environment.
+  set(SQLITE_LDFLAGS ${PLATFORM_LDFLAGS})
+endif()
+
   set(SQLITE_CFLAGS "\
 -DSQLITE_SECURE_DELETE \
 -DSQLITE_ENABLE_COLUMN_METADATA \
@@ -32,9 +45,14 @@ if(UNIX)
 -DSQLITE_MAX_VARIABLE_NUMBER=250000 \
 -fPIC"
   )
-  set(SQLITE_CONFIGURE_ENV
-    ${SQLITE_CONFIGURE_ENV} &&
-    export LDFLAGS=${SQLITE_LDFLAGS} &&
+
+  if(WITH_APPLE_CROSSPLATFORM)
+    set(SQLITE_CFLAGS "${SQLITE_CFLAGS} ${PLATFORM_CFLAGS} -DSQLITE_NOHAVE_SYSTEM=1")
+  endif()
+  
+  set(SQLITE_CONFIGURE_ENV 
+    ${SQLITE_CONFIGURE_ENV} && 
+    export LDFLAGS=${SQLITE_LDFLAGS} && 
     export CFLAGS=${SQLITE_CFLAGS}
   )
   set(SQLITE_CONFIGURATION_ARGS
@@ -62,7 +80,7 @@ ExternalProject_Add(external_sqlite
 
   CONFIGURE_COMMAND ${SQLITE_CONFIGURE_ENV} &&
     cd ${BUILD_DIR}/sqlite/src/external_sqlite/ &&
-    ${CONFIGURE_COMMAND} --prefix=${LIBDIR}/sqlite ${SQLITE_CONFIGURATION_ARGS}
+    ${CONFIGURE_COMMAND} --prefix=${LIBDIR}/sqlite ${SQLITE_CONFIGURATION_ARGS} ${CROSS_COMPILE_FLAGS}
 
   BUILD_COMMAND ${CONFIGURE_ENV} &&
     cd ${BUILD_DIR}/sqlite/src/external_sqlite/ &&
@@ -74,3 +92,8 @@ ExternalProject_Add(external_sqlite
 
   INSTALL_DIR ${LIBDIR}/sqlite
 )
+
+if(WITH_APPLE_CROSSPLATFORM)
+  # Required to provide libs for IOS_PYTHON_STATIC_LIBS
+  harvest_rpath_lib(external_sqlite sqlite/lib sqlite/lib "*.a")
+endif()
