diff --git a/build_files/build_environment/cmake/tiff.cmake b/build_files/build_environment/cmake/tiff.cmake
index 624182b0dbc..60a7a79ba67 100644
--- a/build_files/build_environment/cmake/tiff.cmake
+++ b/build_files/build_environment/cmake/tiff.cmake
@@ -26,6 +26,15 @@ if(APPLE)
   )
 endif()
 
+if(WITH_APPLE_CROSSPLATFORM)
+  # Disable tools build on iOS.
+  set(TIFF_EXTRA_ARGS
+    ${TIFF_EXTRA_ARGS}
+    -Dtiff-tools=OFF
+  )
+endif()
+
+
 ExternalProject_Add(external_tiff
   URL file://${PACKAGE_DIR}/${TIFF_FILE}
   DOWNLOAD_DIR ${DOWNLOAD_DIR}
