diff --git a/build_files/build_environment/cmake/usd.cmake b/build_files/build_environment/cmake/usd.cmake
index 0d28f4f7fa5..283243112cb 100644
--- a/build_files/build_environment/cmake/usd.cmake
+++ b/build_files/build_environment/cmake/usd.cmake
@@ -115,6 +115,33 @@ set(USD_EXTRA_ARGS
   -DSpirvReflect_ROOT=${LIBDIR}/spirv_reflect
 )
 
+if(WITH_APPLE_CROSSPLATFORM)
+
+  # Use iOS utility to set some env vars to help us build for iOS
+  include(cmake/ios_defines.cmake)
+  ios_get_dependency_env_vars(MATERIALX)
+  set(USD_CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${IOSDEP_INCLUDES_STRING}")
+  set(USD_CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${IOSDEP_INCLUDES_STRING}")
+  set(USD_CMAKE_CXX_STANDARD_LIBRARIES "${CMAKE_CXX_STANDARD_LIBRARIES} ${IOSDEP_LIBDIRS_STRING}")
+
+  set(USD_EXTRA_ARGS
+    ${USD_EXTRA_ARGS}
+    # Disable OpenGL for iOS.
+    -DPXR_ENABLE_GL_SUPPORT=OFF
+    # TODO: Re-enable metal support once iOS compatible. Require updating USD.
+    # NOTE: Temporarily disables: PXR_BUILD_GPU_SUPPORT, which will disable hydra storm.
+    -DPXR_ENABLE_METAL_SUPPORT=OFF
+
+    # Override CXX flags to directly provide required headers.	
+    -DCMAKE_CXX_FLAGS=${USD_CMAKE_CXX_FLAGS} 
+    -DCMAKE_C_FLAGS=${USD_CMAKE_C_FLAGS}
+    -DCMAKE_CXX_STANDARD_LIBRARIES=${USD_CMAKE_CXX_STANDARD_LIBRARIES} 
+  )
+  set(USD_BASE_PATCH ${PATCH_DIR}/usd_ios.diff)
+else()
+  set(USD_BASE_PATCH ${PATCH_DIR}/usd.diff)
+endif()
+
 # Ray: I'm not sure if the other platforms relied on this or not but this is no longer
 # needed for windows. If mac/lin confirm, this can be removed.
 if(NOT WIN32)
@@ -137,7 +164,7 @@ ExternalProject_Add(external_usd
     ${USD_EXTRA_PATCHES}
     ${PATCH_CMD} -p 1 -d
       ${BUILD_DIR}/usd/src/external_usd <
-      ${PATCH_DIR}/usd.diff &&
+      ${USD_BASE_PATCH} &&
     ${PATCH_CMD} -p 1 -d
       ${BUILD_DIR}/usd/src/external_usd <
       ${PATCH_DIR}/usd_core_profile.diff &&
