diff --git a/build_files/build_environment/cmake/versions.cmake b/build_files/build_environment/cmake/versions.cmake
index 3fa18afdaf4..51e0986f0ac 100644
--- a/build_files/build_environment/cmake/versions.cmake
+++ b/build_files/build_environment/cmake/versions.cmake
@@ -1432,3 +1432,13 @@ set(HIPRT_FILE hiprt-${HIPRT_VERSION}.tar.gz)
 set(HIPRT_HOMEPAGE https://github.com/GPUOpen-LibrariesAndSDKs/HIPRT)
 set(HIPRT_LICENSE SPDX:MIT)
 set(HIPRT_COPYRIGHT "Copyright (C) 2024 Advanced Micro Devices, Inc. All Rights Reserved. ")
+
+set(LIBB2_VERSION v0.98.1)
+set(LIBB2_LIBRARY_VERSION 02005)
+set(LIBB2_URI https://github.com/BLAKE2/libb2/archive/${LIBB2_VERSION}.tar.gz)
+set(LIBB2_HASH 5ebe2eec2deeb9aedabd58d9cf91d92f)
+set(LIBB2_HASH_TYPE MD5)
+set(LIBB2_FILE libb2-${HIPRT_VERSION}.tar.gz)
+set(LIBB2_HOMEPAGE https://github.com/BLAKE2/libb2)
+set(LIBB2_LICENSE SPDX:CC0-1.0)
+set(LIBB2_COPYRIGHT "<no idea, its CC0>")
