diff --git a/build_files/build_environment/cmake/x264.cmake b/build_files/build_environment/cmake/x264.cmake
index 1485960ba9d..82cb59255b7 100644
--- a/build_files/build_environment/cmake/x264.cmake
+++ b/build_files/build_environment/cmake/x264.cmake
@@ -23,6 +23,13 @@ else()
   set(X264_CONFIGURE_ENV ${CONFIGURE_ENV})
 endif()
 
+if(WITH_APPLE_CROSSPLATFORM)
+  # Disable ASM for iOS for now 
+  # Here seems to have some good tips for building for iOS with asm
+  # https://github.com/kewlbear/x264-ios/blob/master/build-x264.sh
+  set(X264_EXTRA_ARGS ${X264_EXTRA_ARGS} "--disable-asm")
+endif()
+
 ExternalProject_Add(external_x264
   URL file://${PACKAGE_DIR}/${X264_FILE}
   DOWNLOAD_DIR ${DOWNLOAD_DIR}
