diff --git a/source/blender/gpu/metal/mtl_texture.hh b/source/blender/gpu/metal/mtl_texture.hh
index 9e052644058..127c1677eac 100644
--- a/source/blender/gpu/metal/mtl_texture.hh
+++ b/source/blender/gpu/metal/mtl_texture.hh
@@ -19,9 +19,7 @@
 #include <string>
 #include <thread>
 
-#include <Cocoa/Cocoa.h>
 #include <Metal/Metal.h>
-#include <QuartzCore/QuartzCore.h>
 
 @class CAMetalLayer;
 @class MTLCommandQueue;
@@ -594,7 +592,9 @@ inline bool mtl_format_is_writable(MTLPixelFormat format)
     case MTLPixelFormatDepth32Float:
     case MTLPixelFormatDepth32Float_Stencil8:
     case MTLPixelFormatBGR10A2Unorm:
+#if MTL_BACKEND_SUPPORTS_D24_S8_SYMBOLS
     case MTLPixelFormatDepth24Unorm_Stencil8:
+#endif
       return false;
     default:
       return true;
@@ -625,7 +625,9 @@ inline MTLPixelFormat mtl_format_get_writeable_view_format(MTLPixelFormat format
       /* No alternative mirror format. This should not be used for
        * manual data upload */
       return MTLPixelFormatInvalid;
+#if MTL_BACKEND_SUPPORTS_D24_S8_SYMBOLS
     case MTLPixelFormatDepth24Unorm_Stencil8:
+#endif
       /* No direct format, but we'll just mirror the bytes -- `Uint`
        * should ensure bytes are not re-normalized or manipulated */
       // return MTLPixelFormatR32Uint;
@@ -656,8 +658,8 @@ inline MTLTextureUsage mtl_usage_from_gpu(eGPUTextureUsage usage)
   if (usage & GPU_TEXTURE_USAGE_FORMAT_VIEW) {
     mtl_usage = mtl_usage | MTLTextureUsagePixelFormatView;
   }
-#if defined(MAC_OS_VERSION_14_0)
-  if (@available(macOS 14.0, *)) {
+#if defined(MAC_OS_VERSION_14_0) || defined(IPHONE_OS_VERSION_17_0)
+  if (@available(macOS 14.0, ios 17.0, *)) {
     if (usage & GPU_TEXTURE_USAGE_ATOMIC) {
       mtl_usage = mtl_usage | MTLTextureUsageShaderAtomic;
     }
