diff --git a/source/blender/gpu/metal/mtl_texture.mm b/source/blender/gpu/metal/mtl_texture.mm
index c6b8d0da8eb..f3e1ef3d91a 100644
--- a/source/blender/gpu/metal/mtl_texture.mm
+++ b/source/blender/gpu/metal/mtl_texture.mm
@@ -176,9 +176,11 @@ void gpu::MTLTexture::bake_mip_swizzle_view()
     MTLPixelFormat texture_view_pixel_format = gpu_texture_format_to_metal(format_);
     if (texture_view_stencil_) {
       switch (texture_view_pixel_format) {
+#if MTL_BACKEND_SUPPORTS_D24_S8_SYMBOLS
         case MTLPixelFormatDepth24Unorm_Stencil8:
           texture_view_pixel_format = MTLPixelFormatX24_Stencil8;
           break;
+#endif
         case MTLPixelFormatDepth32Float_Stencil8:
           texture_view_pixel_format = MTLPixelFormatX32_Stencil8;
           break;
@@ -1105,6 +1107,7 @@ void gpu::MTLTexture::update_sub(
       [staging_texture release];
     }
 
+#if MTL_BACKEND_SUPPORTS_MANAGED_BUFFERS
     /* Finalize Blit Encoder. */
     if (can_use_direct_blit) {
       /* Textures which use MTLStorageModeManaged need to have updated contents
@@ -1119,11 +1122,11 @@ void gpu::MTLTexture::update_sub(
        * synced back to CPU to avoid an automatic flush overwriting contents. */
       blit_encoder = ctx->main_command_buffer.ensure_begin_blit_encoder();
       if (texture_.storageMode == MTLStorageModeManaged) {
-
         [blit_encoder synchronizeResource:texture_];
       }
       [blit_encoder optimizeContentsForGPUAccess:texture_];
     }
+#endif
 
     /* Decrement texture reference counts. This ensures temporary texture views are released. */
     [texture_handle release];
@@ -1182,11 +1185,11 @@ void MTLTexture::update_sub(int offset[3],
                 destinationSlice:0
                 destinationLevel:0
                destinationOrigin:MTLOriginMake(offset[0], offset[1], 0)];
-
+#if MTL_BACKEND_SUPPORTS_MANAGED_BUFFERS
     if (texture_.storageMode == MTLStorageModeManaged) {
       [blit_encoder synchronizeResource:texture_];
     }
-    [blit_encoder optimizeContentsForGPUAccess:texture_];
+#endif
   }
   else {
     BLI_assert(false);
@@ -1456,13 +1459,15 @@ void gpu::MTLTexture::clear(eGPUDataFormat data_format, const void *data)
       } break;
     }
 
-    /* Textures which use MTLStorageModeManaged need to have updated contents
-     * synced back to CPU to avoid an automatic flush overwriting contents. */
+      /* Textures which use MTLStorageModeManaged need to have updated contents
+       * synced back to CPU to avoid an automatic flush overwriting contents. */
+#if MTL_BACKEND_SUPPORTS_MANAGED_BUFFERS
     id<MTLBlitCommandEncoder> blit_encoder = ctx->main_command_buffer.ensure_begin_blit_encoder();
     if (texture_.storageMode == MTLStorageModeManaged) {
       [blit_encoder synchronizeResource:texture_];
     }
     [blit_encoder optimizeContentsForGPUAccess:texture_];
+#endif
   }
 }
 static MTLTextureSwizzle swizzle_to_mtl(const char swizzle)
@@ -1993,6 +1998,7 @@ void gpu::MTLTexture::read_internal(int mip,
 
     if (copy_successful) {
 
+#if MTL_BACKEND_SUPPORTS_MANAGED_BUFFERS
       /* Use Blit encoder to synchronize results back to CPU. */
       if (dest_buf->get_resource_options() == MTLResourceStorageModeManaged) {
         id<MTLBlitCommandEncoder> enc = ctx->main_command_buffer.ensure_begin_blit_encoder();
@@ -2001,6 +2007,7 @@ void gpu::MTLTexture::read_internal(int mip,
         }
         [enc synchronizeResource:destination_buffer];
       }
+#endif
 
       /* Ensure GPU copy commands have completed. */
       GPU_finish();
@@ -2612,9 +2619,13 @@ void *MTLPixelBuffer::map()
    * in-flight on the GPU. */
   MTLContext *ctx = MTLContext::get();
   BLI_assert(ctx);
+#if MTL_BACKEND_SUPPORTS_MANAGED_BUFFERS
   MTLResourceOptions resource_options = ([ctx->device hasUnifiedMemory]) ?
                                             MTLResourceStorageModeShared :
                                             MTLResourceStorageModeManaged;
+#else
+  MTLResourceOptions resource_options = MTLResourceStorageModeShared;
+#endif
 
   if (buffer_ != nil) {
     id<MTLBuffer> new_buffer = [ctx->device newBufferWithBytes:[buffer_ contents]
@@ -2637,9 +2648,11 @@ void MTLPixelBuffer::unmap()
   }
 
   /* Ensure changes are synchronized. */
+#if MTL_BACKEND_SUPPORTS_MANAGED_BUFFERS
   if (buffer_.resourceOptions & MTLResourceStorageModeManaged) {
     [buffer_ didModifyRange:NSMakeRange(0, size_)];
   }
+#endif
 }
 
 GPUPixelBufferNativeHandle MTLPixelBuffer::get_native_handle()
