diff --git a/source/blender/gpu/metal/mtl_storage_buffer.mm b/source/blender/gpu/metal/mtl_storage_buffer.mm
index d4e074e701c..083b0f88c54 100644
--- a/source/blender/gpu/metal/mtl_storage_buffer.mm
+++ b/source/blender/gpu/metal/mtl_storage_buffer.mm
@@ -372,11 +372,13 @@ void MTLStorageBuf::async_flush_to_host()
   }
 
   /* For discrete memory systems, explicitly flush GPU-resident memory back to host. */
+#if MTL_BACKEND_SUPPORTS_MANAGED_BUFFERS
   id<MTLBuffer> storage_buf_mtl = this->metal_buffer_->get_metal_buffer();
   if (storage_buf_mtl.storageMode == MTLStorageModeManaged) {
     id<MTLBlitCommandEncoder> blit_encoder = ctx->main_command_buffer.ensure_begin_blit_encoder();
     [blit_encoder synchronizeResource:storage_buf_mtl];
   }
+#endif
 
   /* Encode event signal and flush command buffer to ensure GPU work is in the pipeline for future
    * reads. */
@@ -419,9 +421,11 @@ void MTLStorageBuf::read(void *data)
                         toBuffer:staging_buf_mtl
                destinationOffset:0
                             size:size_in_bytes_];
+#if MTL_BACKEND_SUPPORTS_MANAGED_BUFFERS
     if (staging_buf_mtl.storageMode == MTLStorageModeManaged) {
       [blit_encoder synchronizeResource:staging_buf_mtl];
     }
+#endif
 
     /* Device-only reads will always stall the GPU pipe. */
     GPU_finish();
@@ -448,6 +452,7 @@ void MTLStorageBuf::read(void *data)
       GPU_finish();
     }
 
+#if MTL_BACKEND_SUPPORTS_MANAGED_BUFFERS
     /* Managed buffers need to be explicitly flushed back to host. */
     if (metal_buffer_->get_resource_options() & MTLResourceStorageModeManaged) {
       /* Fetch active context. */
@@ -462,6 +467,7 @@ void MTLStorageBuf::read(void *data)
       /* Wait for the blit to finish. */
       GPU_finish();
     }
+#endif
 
     /* Read data. NOTE: Unless explicitly synchronized with GPU work, results may not be ready. */
     memcpy(data, metal_buffer_->get_host_ptr(), size_in_bytes_);
