diff --git a/source/blender/gpu/metal/mtl_texture_util.mm b/source/blender/gpu/metal/mtl_texture_util.mm
index e13a59ee7bb..26caaff433a 100644
--- a/source/blender/gpu/metal/mtl_texture_util.mm
+++ b/source/blender/gpu/metal/mtl_texture_util.mm
@@ -34,16 +34,184 @@ namespace blender::gpu {
 
 MTLPixelFormat gpu_texture_format_to_metal(TextureFormat tex_format)
 {
-#define CASE(a, b, c, blender_enum, d, mtl_pixel_enum, f, g, h) \
-  case TextureFormat::blender_enum: \
-    return MTLPixelFormat##mtl_pixel_enum;
-
   switch (tex_format) {
-    GPU_TEXTURE_FORMAT_EXPAND(CASE)
+    /* Texture & Render-Buffer Formats. */
+    case TextureFormat::UINT_8_8_8_8:
+      return MTLPixelFormatRGBA8Uint;
+    case TextureFormat::SINT_8_8_8_8:
+      return MTLPixelFormatRGBA8Sint;
+    case TextureFormat::UNORM_8_8_8_8:
+      return MTLPixelFormatRGBA8Unorm;
+    case TextureFormat::UINT_32_32_32_32:
+      return MTLPixelFormatRGBA32Uint;
+    case TextureFormat::SINT_32_32_32_32:
+      return MTLPixelFormatRGBA32Sint;
+    case TextureFormat::SFLOAT_32_32_32_32:
+      return MTLPixelFormatRGBA32Float;
+    case TextureFormat::UINT_16_16_16_16:
+      return MTLPixelFormatRGBA16Uint;
+    case TextureFormat::SINT_16_16_16_16:
+      return MTLPixelFormatRGBA16Sint;
+    case TextureFormat::SFLOAT_16_16_16_16:
+      return MTLPixelFormatRGBA16Float;
+    case TextureFormat::UNORM_16_16_16_16:
+      return MTLPixelFormatRGBA16Unorm;
+    case TextureFormat::UINT_8_8:
+      return MTLPixelFormatRG8Uint;
+    case TextureFormat::SINT_8_8:
+      return MTLPixelFormatRG8Sint;
+    case TextureFormat::UNORM_8_8:
+      return MTLPixelFormatRG8Unorm;
+    case TextureFormat::UINT_32_32:
+      return MTLPixelFormatRG32Uint;
+    case TextureFormat::SINT_32_32:
+      return MTLPixelFormatRG32Sint;
+    case TextureFormat::SFLOAT_32_32:
+      return MTLPixelFormatRG32Float;
+    case TextureFormat::UINT_16_16:
+      return MTLPixelFormatRG16Uint;
+    case TextureFormat::SINT_16_16:
+      return MTLPixelFormatRG16Sint;
+    case TextureFormat::SFLOAT_16_16:
+      return MTLPixelFormatRG16Float;
+    case TextureFormat::UNORM_16_16:
+      return MTLPixelFormatRG16Unorm;
+    case TextureFormat::UINT_8:
+      return MTLPixelFormatR8Uint;
+    case TextureFormat::SINT_8:
+      return MTLPixelFormatR8Sint;
+    case TextureFormat::UNORM_8:
+      return MTLPixelFormatR8Unorm;
+    case TextureFormat::UINT_32:
+      return MTLPixelFormatR32Uint;
+    case TextureFormat::SINT_32:
+      return MTLPixelFormatR32Sint;
+    case TextureFormat::SFLOAT_32:
+      return MTLPixelFormatR32Float;
+    case TextureFormat::UINT_16:
+      return MTLPixelFormatR16Uint;
+    case TextureFormat::SINT_16:
+      return MTLPixelFormatR16Sint;
+    case TextureFormat::SFLOAT_16:
+      return MTLPixelFormatR16Float;
+    case TextureFormat::UNORM_16:
+      return MTLPixelFormatR16Unorm;
+    /* Special formats texture & render-buffer. */
+    case TextureFormat::UNORM_10_10_10_2:
+      return MTLPixelFormatRGB10A2Unorm;
+    case TextureFormat::UINT_10_10_10_2:
+      return MTLPixelFormatRGB10A2Uint;
+    case TextureFormat::UFLOAT_11_11_10:
+      return MTLPixelFormatRG11B10Float;
+    case TextureFormat::SFLOAT_32_DEPTH_UINT_8:
+      return MTLPixelFormatDepth32Float_Stencil8;
+    case TextureFormat::SRGBA_8_8_8_8:
+      return MTLPixelFormatRGBA8Unorm_sRGB;
+    /* Texture only formats. */
+    case TextureFormat::SFLOAT_16_16_16:
+      /* 48-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA16Float;
+    case TextureFormat::SNORM_16_16_16_16:
+      return MTLPixelFormatRGBA16Snorm;
+    case TextureFormat::SNORM_8_8_8_8:
+      return MTLPixelFormatRGBA8Snorm;
+    case TextureFormat::SFLOAT_32_32_32:
+      /* 96-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA32Float;
+    case TextureFormat::SINT_32_32_32:
+      /* 96-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA32Sint;
+    case TextureFormat::UINT_32_32_32:
+      /* 96-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA32Uint;
+    case TextureFormat::SNORM_16_16_16:
+      /* 48-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA16Snorm;
+    case TextureFormat::SINT_16_16_16:
+      /* 48-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA16Sint;
+    case TextureFormat::UINT_16_16_16:
+      /* 48-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA16Uint;
+    case TextureFormat::UNORM_16_16_16:
+      /* 48-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA16Unorm;
+    case TextureFormat::SNORM_8_8_8:
+      /* 24-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA8Snorm;
+    case TextureFormat::UNORM_8_8_8:
+      /* 24-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA8Unorm;
+    case TextureFormat::SINT_8_8_8:
+      /* 24-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA8Sint;
+    case TextureFormat::UINT_8_8_8:
+      /* 24-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA8Uint;
+    case TextureFormat::SNORM_16_16:
+      return MTLPixelFormatRG16Snorm;
+    case TextureFormat::SNORM_8_8:
+      return MTLPixelFormatRG8Snorm;
+    case TextureFormat::SNORM_16:
+      return MTLPixelFormatR16Snorm;
+    case TextureFormat::SNORM_8:
+      return MTLPixelFormatR8Snorm;
+    /* Special formats, texture only. */
+    case TextureFormat::SRGB_DXT1:
+      if (@available(iOS 16.4, *)) {
+        return MTLPixelFormatBC1_RGBA_sRGB;
+      }
+      else {
+        // Fallback on earlier versions
+      }
+    case TextureFormat::SRGB_DXT3:
+      if (@available(iOS 16.4, *)) {
+        return MTLPixelFormatBC2_RGBA_sRGB;
+      }
+      else {
+        // Fallback on earlier versions
+      }
+    case TextureFormat::SRGB_DXT5:
+      if (@available(iOS 16.4, *)) {
+        return MTLPixelFormatBC3_RGBA_sRGB;
+      }
+      else {
+        // Fallback on earlier versions
+      }
+    case TextureFormat::SNORM_DXT1:
+      if (@available(iOS 16.4, *)) {
+        return MTLPixelFormatBC1_RGBA;
+      }
+      else {
+        // Fallback on earlier versions
+      }
+    case TextureFormat::SNORM_DXT3:
+      if (@available(iOS 16.4, *)) {
+        return MTLPixelFormatBC2_RGBA;
+      }
+      else {
+        // Fallback on earlier versions
+      }
+    case TextureFormat::SNORM_DXT5:
+      if (@available(iOS 16.4, *)) {
+        return MTLPixelFormatBC3_RGBA;
+      }
+      else {
+        // Fallback on earlier versions
+      }
+    case TextureFormat::SRGBA_8_8_8:
+      /* 24-Bit pixel format are not supported. Emulate using a padded type with alpha. */
+      return MTLPixelFormatRGBA8Unorm_sRGB;
+    case TextureFormat::UFLOAT_9_9_9_EXP_5:
+      return MTLPixelFormatRGB9E5Float;
+    /* Depth Formats. */
+    case TextureFormat::SFLOAT_32_DEPTH:
+      return MTLPixelFormatDepth32Float;
+    case TextureFormat::UNORM_16_DEPTH:
+      return MTLPixelFormatDepth16Unorm;
     case TextureFormat::Invalid:
-      break;
+      BLI_assert_msg(false, "Unrecognised GPU pixel format!\n");
   }
-#undef CASE
   BLI_assert_msg(false, "Unrecognised GPU pixel format!\n");
   return MTLPixelFormatRGBA8Unorm;
 }
@@ -105,7 +273,9 @@ size_t get_mtl_format_bytesize(MTLPixelFormat tex_format)
       return 8;
     case MTLPixelFormatRGBA8Unorm_sRGB:
     case MTLPixelFormatDepth32Float:
+#if MTL_BACKEND_SUPPORTS_D24_S8_SYMBOLS
     case MTLPixelFormatDepth24Unorm_Stencil8:
+#endif
       return 4;
     case MTLPixelFormatDepth16Unorm:
       return 2;
@@ -182,7 +352,9 @@ int get_mtl_format_num_components(MTLPixelFormat tex_format)
     case MTLPixelFormatR16Snorm:
     case MTLPixelFormatDepth32Float:
     case MTLPixelFormatDepth16Unorm:
+#if MTL_BACKEND_SUPPORTS_D24_S8_SYMBOLS
     case MTLPixelFormatDepth24Unorm_Stencil8:
+#endif
       /* Treating this format as single-channel for direct data copies -- Stencil component is not
        * addressable. */
       return 1;
