diff --git a/source/blender/gpu/metal/mtl_query.mm b/source/blender/gpu/metal/mtl_query.mm
index c8cc18e2023..1e5e13766ad 100644
--- a/source/blender/gpu/metal/mtl_query.mm
+++ b/source/blender/gpu/metal/mtl_query.mm
@@ -102,6 +102,7 @@ void MTLQueryPool::get_occlusion_result(MutableSpan<uint32_t> r_values)
 
   /* Create a blit encoder to synchronize the query buffer results between
    * GPU and CPU when not using shared-memory. */
+#if MTL_BACKEND_SUPPORTS_MANAGED_BUFFERS
   if ([ctx->device hasUnifiedMemory] == false) {
     id<MTLBlitCommandEncoder> blit_encoder = ctx->main_command_buffer.ensure_begin_blit_encoder();
     BLI_assert(blit_encoder);
@@ -110,6 +111,7 @@ void MTLQueryPool::get_occlusion_result(MutableSpan<uint32_t> r_values)
     }
     BLI_assert(ctx->get_inside_frame());
   }
+#endif
 
   /* Wait for GPU operations to complete and for query buffer contents
    * to be synchronized back to host memory. */
