diff --git a/source/blender/gpu/metal/mtl_command_buffer.mm b/source/blender/gpu/metal/mtl_command_buffer.mm
index b736cc34617..e885ad6149a 100644
--- a/source/blender/gpu/metal/mtl_command_buffer.mm
+++ b/source/blender/gpu/metal/mtl_command_buffer.mm
@@ -10,7 +10,11 @@
 #include "mtl_debug.hh"
 #include "mtl_framebuffer.hh"
 
-#include "intern/GHOST_ContextMTL.hh"
+#ifdef WITH_APPLE_CROSSPLATFORM
+#  include "intern/GHOST_ContextIOS.hh"
+#else
+#  include "intern/GHOST_ContextMTL.hh"
+#endif
 
 #include <fstream>
 
@@ -51,7 +55,7 @@ id<MTLCommandBuffer> MTLCommandBufferManager::ensure_begin()
      * NOTE: We currently stall until completion of GPU work upon ::submit if we have reached the
      * in-flight command buffer limit. */
     BLI_assert(MTLCommandBufferManager::num_active_cmd_bufs_in_system <
-               GHOST_ContextMTL::max_command_buffer_count);
+               GHOST_ContextMetal::max_command_buffer_count);
 
     if (G.debug & G_DEBUG_GPU) {
       /* Debug: Enable Advanced Errors for GPU work execution. */
@@ -138,12 +142,12 @@ bool MTLCommandBufferManager::submit(bool wait)
   /* If we have too many active command buffers in flight, wait until completed to avoid running
    * out. We can increase */
   if (MTLCommandBufferManager::num_active_cmd_bufs_in_system >=
-      (GHOST_ContextMTL::max_command_buffer_count - 1))
+      (GHOST_ContextMetal::max_command_buffer_count - 1))
   {
     wait = true;
     MTL_LOG_WARNING(
         "Maximum number of command buffers in flight. Host will wait until GPU work has "
-        "completed. Consider increasing GHOST_ContextMTL::max_command_buffer_count or reducing "
+        "completed. Consider increasing GHOST_Context*::max_command_buffer_count or reducing "
         "work fragmentation to better utilize system hardware. Command buffers are flushed upon "
         "GPUContext switches, this is the most common cause of excessive command buffer "
         "generation.");
@@ -602,8 +606,12 @@ bool MTLCommandBufferManager::insert_memory_barrier(eGPUBarrier barrier_bits,
   /* Resolve scope. */
   MTLBarrierScope scope = 0;
   if (barrier_bits & GPU_BARRIER_SHADER_IMAGE_ACCESS || barrier_bits & GPU_BARRIER_TEXTURE_FETCH) {
+#if MTL_BACKEND_SUPPORTS_RENDER_TARGET_BARRIER
     bool is_compute = (active_command_encoder_type_ != MTL_RENDER_COMMAND_ENCODER);
     scope |= (is_compute ? 0 : MTLBarrierScopeRenderTargets) | MTLBarrierScopeTextures;
+#else
+    scope |= MTLBarrierScopeTextures;
+#endif
   }
   if (barrier_bits & GPU_BARRIER_SHADER_STORAGE ||
       barrier_bits & GPU_BARRIER_VERTEX_ATTRIB_ARRAY || barrier_bits & GPU_BARRIER_ELEMENT_ARRAY ||
@@ -642,19 +650,23 @@ bool MTLCommandBufferManager::insert_memory_barrier(eGPUBarrier barrier_bits,
           after_stage_flags = MTLRenderStageFragment;
         }
 
+#if MTL_BACKEND_SUPPORTS_RENDER_TARGET_BARRIER
         id<MTLRenderCommandEncoder> rec = this->get_active_render_command_encoder();
         BLI_assert(rec != nil);
         [rec memoryBarrierWithScope:scope
                         afterStages:after_stage_flags
                        beforeStages:before_stage_flags];
+#endif
         return true;
       }
 
       /* Compute. */
       case MTL_COMPUTE_COMMAND_ENCODER: {
+#if MTL_BACKEND_SUPPORTS_RENDER_TARGET_BARRIER
         id<MTLComputeCommandEncoder> rec = this->get_active_compute_command_encoder();
         BLI_assert(rec != nil);
         [rec memoryBarrierWithScope:scope];
+#endif
         return true;
       }
     }
