diff --git a/source/blender/gpu/metal/mtl_context.hh b/source/blender/gpu/metal/mtl_context.hh
index e6ef64bce2d..a30223f68fd 100644
--- a/source/blender/gpu/metal/mtl_context.hh
+++ b/source/blender/gpu/metal/mtl_context.hh
@@ -18,7 +18,11 @@
 #endif
 
 #include "intern/GHOST_Context.hh"
-#include "intern/GHOST_ContextMTL.hh"
+#ifdef WITH_APPLE_CROSSPLATFORM
+#  include "intern/GHOST_ContextIOS.hh"
+#else
+#  include "intern/GHOST_ContextMTL.hh"
+#endif
 #include "intern/GHOST_Window.hh"
 
 #include "mtl_backend.hh"
@@ -30,9 +34,7 @@
 #include "mtl_shader_interface.hh"
 #include "mtl_texture.hh"
 
-#include <Cocoa/Cocoa.h>
 #include <Metal/Metal.h>
-#include <QuartzCore/QuartzCore.h>
 #include <chrono>
 #include <mutex>
 
@@ -728,7 +730,7 @@ class MTLContext : public Context {
 
  private:
   /* Parent Context. */
-  GHOST_ContextMTL *ghost_context_;
+  GHOST_ContextMetal *ghost_context_;
 
   /* Render Passes and Frame-buffers. */
   id<MTLTexture> default_fbo_mtltexture_ = nil;
