diff --git a/source/blender/gpu/metal/mtl_immediate.hh b/source/blender/gpu/metal/mtl_immediate.hh
index dc78c4cd928..28998a23333 100644
--- a/source/blender/gpu/metal/mtl_immediate.hh
+++ b/source/blender/gpu/metal/mtl_immediate.hh
@@ -14,11 +14,7 @@
 
 #include "gpu_immediate_private.hh"
 
-#include "mtl_context.hh"
-
-#include <Cocoa/Cocoa.h>
 #include <Metal/Metal.h>
-#include <QuartzCore/QuartzCore.h>
 
 namespace blender::gpu {
 
