diff --git a/intern/cycles/cmake/external_libs.cmake b/intern/cycles/cmake/external_libs.cmake
index 647861af0b1..6dc64f34875 100644
--- a/intern/cycles/cmake/external_libs.cmake
+++ b/intern/cycles/cmake/external_libs.cmake
@@ -76,21 +76,26 @@ endif()
 ###########################################################################
 
 if(WITH_CYCLES_DEVICE_METAL)
-  find_library(METAL_LIBRARY Metal)
-
-  # This file was added in the 12.0 SDK, use it as a way to detect the version.
-  if(METAL_LIBRARY)
-    if(EXISTS "${METAL_LIBRARY}/Headers/MTLFunctionStitching.h")
-      set(METAL_FOUND ON)
-    else()
-      message(STATUS "Metal version too old, must be SDK 12.0 or newer")
-      set(METAL_FOUND OFF)
+  if(WITH_APPLE_CROSSPLATFORM)
+    # We will use platform Metal framework for Apple cross-platform builds (iOS).
+    set(METAL_FOUND ON)
+  else()
+    find_library(METAL_LIBRARY Metal)
+
+    # This file was added in the 12.0 SDK, use it as a way to detect the version.
+    if(METAL_LIBRARY)
+      if(EXISTS "${METAL_LIBRARY}/Headers/MTLFunctionStitching.h")
+        set(METAL_FOUND ON)
+      else()
+        message(STATUS "Metal version too old, must be SDK 12.0 or newer")
+        set(METAL_FOUND OFF)
+      endif()
     endif()
-  endif()
 
-  set_and_warn_library_found("Metal" METAL_FOUND WITH_CYCLES_DEVICE_METAL)
-  if(METAL_FOUND)
-    message(STATUS "Found Metal: ${METAL_LIBRARY}")
+    set_and_warn_library_found("Metal" METAL_FOUND WITH_CYCLES_DEVICE_METAL)
+    if(METAL_FOUND)
+      message(STATUS "Found Metal: ${METAL_LIBRARY}")
+    endif()
   endif()
 endif()
 
