diff --git a/source/blender/editors/interface/view2d/view2d_ops.cc b/source/blender/editors/interface/view2d/view2d_ops.cc
index e5802d4eb20..685b1234918 100644
--- a/source/blender/editors/interface/view2d/view2d_ops.cc
+++ b/source/blender/editors/interface/view2d/view2d_ops.cc
@@ -231,6 +231,17 @@ static wmOperatorStatus view_pan_exec(bContext *C, wmOperator *op)
 /* set up modal operator and relevant settings */
 static wmOperatorStatus view_pan_invoke(bContext *C, wmOperator *op, const wmEvent *event)
 {
+#ifdef WITH_APPLE_CROSSPLATFORM
+  ScrArea *area = CTX_wm_area(C);
+  ARegion *region = CTX_wm_region(C);
+  /* Only scroll the File/Asset Browser and Asset Shelf views with multiple finger on iOS. */
+  if ((area->spacetype == SPACE_FILE || region->regiontype == RGN_TYPE_ASSET_SHELF) &&
+      !(event->flag & WM_EVENT_MULTITOUCH_TWO_FINGERS))
+  {
+    return OPERATOR_FINISHED;
+  }
+#endif
+
   wmWindow *window = CTX_wm_window(C);
 
   /* set up customdata */
