diff --git a/source/blender/windowmanager/intern/wm.cc b/source/blender/windowmanager/intern/wm.cc
index c2296bba05e..6cf3eda076a 100644
--- a/source/blender/windowmanager/intern/wm.cc
+++ b/source/blender/windowmanager/intern/wm.cc
@@ -582,24 +582,32 @@ void wm_close_and_free(bContext *C, wmWindowManager *wm)
   MEM_delete(wm->runtime);
 }
 
-void WM_main(bContext *C)
+void WM_main_entry(bContext *C)
 {
   /* Single refresh before handling events.
    * This ensures we don't run operators before the depsgraph has been evaluated. */
   wm_event_do_refresh_wm_and_depsgraph(C);
+}
 
-  while (true) {
+void WM_main_loop_body(bContext *C)
+{
+  /* Get events from ghost, handle window events, add to window queues. */
+  wm_window_events_process(C);
 
-    /* Get events from ghost, handle window events, add to window queues. */
-    wm_window_events_process(C);
+  /* Per window, all events to the window, screen, area and region handlers. */
+  wm_event_do_handlers(C);
 
-    /* Per window, all events to the window, screen, area and region handlers. */
-    wm_event_do_handlers(C);
+  /* Events have left notes about changes, we handle and cache it. */
+  wm_event_do_notifiers(C);
 
-    /* Events have left notes about changes, we handle and cache it. */
-    wm_event_do_notifiers(C);
+  /* Execute cached changes draw. */
+  wm_draw_update(C);
+}
 
-    /* Execute cached changes draw. */
-    wm_draw_update(C);
+void WM_main(bContext *C)
+{
+  WM_main_entry(C);
+  while (true) {
+    WM_main_loop_body(C);
   }
 }
