diff --git a/source/blender/windowmanager/intern/wm_event_system.cc b/source/blender/windowmanager/intern/wm_event_system.cc
index 5e19deacb23..46a8fefcce5 100644
--- a/source/blender/windowmanager/intern/wm_event_system.cc
+++ b/source/blender/windowmanager/intern/wm_event_system.cc
@@ -5440,6 +5440,15 @@ static wmEventType wm_event_type_from_ghost_key(GHOST_TKey key)
     case GHOST_kKeyUnknown:
       return EVT_UNKNOWNKEY;
 
+#if (WITH_APPLE_CROSSPLATFORM)
+      /* IOS_FIXME - Event to get multi text edit events from iOS into Blender */
+    case GHOST_kKeyTextEdit:
+      return EVT_TEXTEDIT;
+
+#else
+    case GHOST_kKeyF24:
+#endif
+
 #if defined(__GNUC__) || defined(__clang__)
       /* Ensure all members of this enum are handled, otherwise generate a compiler warning.
        * Note that these members have been handled, these ranges are to satisfy the compiler. */
@@ -6050,6 +6059,13 @@ void wm_event_add_ghostevent(wmWindowManager *wm,
         event.flag |= WM_EVENT_SCROLL_INVERT;
       }
 
+      /* Distinguish between single finger and multi-finger events for iOS.*/
+      if (pd->numFingers == 2) {
+        event.flag |= WM_EVENT_MULTITOUCH_TWO_FINGERS;
+      }
+      else if (pd->numFingers == 3) {
+        event.flag |= WM_EVENT_MULTITOUCH_THREE_FINGERS;
+      }
 #if !defined(WIN32) && !defined(__APPLE__)
       /* Ensure "auto" is used when supported. */
       char trackpad_scroll_direction = U.trackpad_scroll_direction;
@@ -6070,6 +6086,40 @@ void wm_event_add_ghostevent(wmWindowManager *wm,
       wm_event_add_trackpad(win, &event, delta[0], delta[1]);
       break;
     }
+    /* Multi-touch. */
+    case GHOST_kEventTouch: {
+      const GHOST_TEventTouchData *td = static_cast<const GHOST_TEventTouchData *>(customdata);
+      switch (td->subtype) {
+        case GHOST_kTouchEventEdgeSwipeInLeft:
+          event.type = TOUCH_EDGE_SWIPE_IN_LEFT;
+          event.val = KM_PRESS;
+          wm_event_add_intern(win, &event);
+          break;
+        case GHOST_kTouchEventEdgeSwipeInRight:
+          event.type = TOUCH_EDGE_SWIPE_IN_RIGHT;
+          event.val = KM_PRESS;
+          wm_event_add_intern(win, &event);
+          break;
+        default:
+          break;
+      }
+      break;
+    }
+    case GHOST_kEventTwoFingerTap:
+      event.type = TOUCH_TWO_FINGER_TAP;
+      event.val = KM_PRESS;
+      wm_event_add_intern(win, &event);
+      break;
+    case GHOST_kEventThreeFingerTap:
+      event.type = TOUCH_THREE_FINGER_TAP;
+      event.val = KM_PRESS;
+      wm_event_add_intern(win, &event);
+      break;
+    case GHOST_kEventFourFingerTap:
+      event.type = TOUCH_FOUR_FINGER_TAP;
+      event.val = KM_PRESS;
+      wm_event_add_intern(win, &event);
+      break;
     /* Mouse button. */
     case GHOST_kEventButtonDown:
     case GHOST_kEventButtonUp: {
