diff --git a/source/blender/windowmanager/intern/wm_keymap.cc b/source/blender/windowmanager/intern/wm_keymap.cc
index 015053c9a14..da810a390ff 100644
--- a/source/blender/windowmanager/intern/wm_keymap.cc
+++ b/source/blender/windowmanager/intern/wm_keymap.cc
@@ -234,6 +234,9 @@ int WM_keymap_item_map_type_get(const wmKeyMapItem *kmi)
   if (ISNDOF(kmi->type)) {
     return KMI_TYPE_NDOF;
   }
+  if (ISTOUCH(kmi->type)) {
+    return KMI_TYPE_TOUCH;
+  }
   if (kmi->type == KM_TEXTINPUT) {
     return KMI_TYPE_TEXTINPUT;
   }
