diff --git a/scripts/startup/bl_operators/__init__.py b/scripts/startup/bl_operators/__init__.py
index a9331415626..5202adef14b 100644
--- a/scripts/startup/bl_operators/__init__.py
+++ b/scripts/startup/bl_operators/__init__.py
@@ -35,6 +35,7 @@ _modules = [
     "screen_play_rendered_anim",
     "sequencer",
     "spreadsheet",
+    "touch",
     "userpref",
     "uvcalc_follow_active",
     "uvcalc_lightmap",
