diff --git a/source/blender/editors/space_view3d/view3d_navigate_view_rotate.cc b/source/blender/editors/space_view3d/view3d_navigate_view_rotate.cc
index 3d79bd7baec..26ac5553a6b 100644
--- a/source/blender/editors/space_view3d/view3d_navigate_view_rotate.cc
+++ b/source/blender/editors/space_view3d/view3d_navigate_view_rotate.cc
@@ -367,6 +367,13 @@ static wmOperatorStatus viewrotate_invoke_impl(bContext *C,
 
 static wmOperatorStatus viewrotate_invoke(bContext *C, wmOperator *op, const wmEvent *event)
 {
+#ifdef WITH_APPLE_CROSSPLATFORM
+  /* Only scroll view with multiple fingers on iOS. */
+  if (!(event->flag & WM_EVENT_MULTITOUCH_TWO_FINGERS)) {
+    return OPERATOR_FINISHED;
+  }
+#endif
+
   return view3d_navigate_invoke_impl(C, op, event, &ViewOpsType_rotate);
 }
 
