diff --git a/scripts/startup/bl_ui/space_userpref.py b/scripts/startup/bl_ui/space_userpref.py
index ffed0378d41..341f4e5e56f 100644
--- a/scripts/startup/bl_ui/space_userpref.py
+++ b/scripts/startup/bl_ui/space_userpref.py
@@ -701,6 +701,8 @@ class USERPREF_PT_system_display_graphics(SystemPanel, CenterAlignMixIn, Panel):
     @classmethod
     def poll(cls, _context):
         import platform
+        # is_apple_device = (platform.system() == 'Darwin') or (platform.system() == 'iOS') or (platform.system() == 'iPadOS')
+        # return !is_apple_device
         return platform.system() != 'Darwin'
 
     def draw_centered(self, context, layout):
@@ -734,7 +736,7 @@ class USERPREF_PT_system_os_settings(SystemPanel, CenterAlignMixIn, Panel):
     def poll(cls, _context):
         # macOS isn't supported.
         from sys import platform
-        if platform == "darwin":
+        if platform == "darwin" or platform == "ios":
             return False
         return True
 
@@ -828,7 +830,7 @@ class USERPREF_PT_system_memory(SystemPanel, CenterAlignMixIn, Panel):
         col.prop(system, "vbo_time_out", text="VBO Time Out")
         col.prop(system, "vbo_collection_rate", text="Garbage Collection Rate")
 
-        if sys.platform != "darwin":
+        if sys.platform != "darwin" and sys.platform != "ios":
             layout.separator()
             col = layout.column(align=True)
             col.active = system.gpu_backend != 'VULKAN'
