diff --git a/scripts/startup/bl_ui/space_topbar.py b/scripts/startup/bl_ui/space_topbar.py
index 208d13a0354..e4daf62b90a 100644
--- a/scripts/startup/bl_ui/space_topbar.py
+++ b/scripts/startup/bl_ui/space_topbar.py
@@ -169,8 +169,14 @@ class TOPBAR_MT_file(Menu):
 
         layout.separator()
 
-        layout.operator_context = 'EXEC_AREA' if context.blend_data.is_saved else 'INVOKE_AREA'
-        layout.operator("wm.save_mainfile", text="Save", icon='FILE_TICK')
+        # iOS Workaround: Provide a clean alternative to write errors caused by sandbox restrictions
+        import os
+        file_is_writable = os.access(bpy.data.filepath, os.W_OK)
+
+        sub = layout.row()
+        sub.enabled = file_is_writable
+        sub.operator_context = 'EXEC_AREA' if context.blend_data.is_saved else 'INVOKE_AREA'
+        sub.operator("wm.save_mainfile", text="Save", icon='FILE_TICK')
 
         layout.operator_context = 'INVOKE_AREA'
         layout.operator("wm.save_as_mainfile", text="Save As...")
@@ -178,7 +184,7 @@ class TOPBAR_MT_file(Menu):
         layout.operator("wm.save_as_mainfile", text="Save Copy...").copy = True
 
         sub = layout.row()
-        sub.enabled = context.blend_data.is_saved
+        sub.enabled = context.blend_data.is_saved and file_is_writable
         sub.operator_context = 'EXEC_AREA'
         sub.operator("wm.save_mainfile", text="Save Incremental").incremental = True
 
