diff --git a/source/blender/makesdna/intern/CMakeLists.txt b/source/blender/makesdna/intern/CMakeLists.txt
index cf5e4b26ccf..a4c02100ba9 100644
--- a/source/blender/makesdna/intern/CMakeLists.txt
+++ b/source/blender/makesdna/intern/CMakeLists.txt
@@ -73,27 +73,29 @@ set(SRC
   ${dna_header_string_file}
 )
 
-# BLI_mempool.cc needs these files for ThreadMutex when
-# compiling in address sanitizer mode.
-if(WITH_COMPILER_ASAN)
-  list(APPEND SRC
-  )
-endif()
+if(NOT WITH_CROSSCOMPILED_TOOLS)
+  # BLI_mempool.cc needs these files for ThreadMutex when
+  # compiling in address sanitizer mode.
+  if(WITH_COMPILER_ASAN)
+    list(APPEND SRC
+    )
+  endif()
 
 # `SRC_DNA_INC` is defined in the parent directory.
 
 add_cc_flags_custom_test(makesdna)
 
-add_executable(makesdna ${SRC} ${SRC_DNA_INC})
-setup_platform_linker_flags(makesdna)
-setup_platform_linker_libs(makesdna)
-blender_target_include_dirs(makesdna ${INC})
-blender_target_include_dirs_sys(makesdna ${INC_SYS})
-target_link_libraries(makesdna ${LIB})
-
-if(WIN32 AND NOT UNIX)
-  if(DEFINED PTHREADS_LIBRARIES)
-    target_link_libraries(makesdna PRIVATE ${PTHREADS_LIBRARIES})
+  add_executable(makesdna ${SRC} ${SRC_DNA_INC})
+  setup_platform_linker_flags(makesdna)
+  setup_platform_linker_libs(makesdna)
+  blender_target_include_dirs(makesdna ${INC})
+  blender_target_include_dirs_sys(makesdna ${INC_SYS})
+  target_link_libraries(makesdna ${LIB})
+
+  if(WIN32 AND NOT UNIX)
+    if(DEFINED PTHREADS_LIBRARIES)
+      target_link_libraries(makesdna PRIVATE ${PTHREADS_LIBRARIES})
+    endif()
   endif()
 endif()
 
